### PR TITLE
APP-2044 Kill test-e2e child process on exit except in Github Workflow

### PR DESCRIPTION
**Pull request by ethanlook**
This updates `e2e.sh` to kill zombie child processes at the end of the script.

Killing processes using `pkill` is not allowed in CI without `sudo`. The target does not use `sudo` so someone running it locally does not require the user's password. Instead, the flag `-k` is used in the Github Workflow (and passed down in the makefile target definition) to avoid calling `pkill`.

The default to kill the child process must first be merged as false so the workflow still passes. I'll follow up this PR with another PR to flip the default to true.

This change was [validated](https://github.com/viamrobotics/rdk/pull/2553/checks) by using an [empty commit/PR](https://github.com/viamrobotics/rdk/pull/2553) against the branch. I'm currently [documenting the process](https://viam.atlassian.net/wiki/spaces/ENG/pages/207257613/Engineering+Processes+and+Common+Workflows#Updating-GitHub-Workflows).

### CODE REVIEW

**Comment by ethanlook** [comment]
Before merging, this commit will be removed: [DO NOT MERGE: change branch for E2E test](https://github.com/viamrobotics/rdk/pull/2554/commits/fd7d99e586ac97c1f9d185ef004f264f5dd705fc)

**Comment by ethanlook** [comment]
> Before merging, this commit will be removed: [DO NOT MERGE: change branch for E2E test](https://github.com/viamrobotics/rdk/pull/2554/commits/fd7d99e586ac97c1f9d185ef004f264f5dd705fc)

I removed this commit.

**Comment by ethanlook** [comment]
The default to kill the child process must first be merged as false so the workflow still passes. I'll follow up this PR with another PR to flip the default to true.

**Review by michaellee1019 — APPROVED** [review]

**Pull request merged**
